lib/widgets/new_expense.dart

```diff
@@ -0,0 +1,256 @@
+        //Sets the decoration, in this case the label
+        label: Text('Title'),
+      ),
+    );
+
+    var amountComponent = Expanded(
+      //Makes the component take up all the remaining space
+      child: TextField(
+        //Creates a text field
+        controller: _amountController, //Sets the controller
+        keyboardType: TextInputType.number,
+        decoration: const InputDecoration(
+          //Sets the decoration, in this case the label
+          prefixText: '\$ ', //Adds a prefix to the input
+          label: Text('Amount'),
+        ),
+      ),
+    );
+
+    var dropDownButton = DropdownButton(
+        //Creates a dropdown button
+        value: _selectedCategory, //Sets the selected value
+        items: Category.values //Creates the items
+            .map((category) => DropdownMenuItem(
+                  //Creates a dropdown menu item
+                  value: category,
+                  child: Text(category.name.toUpperCase()),
+                ))
+            .toList(), //Converts the iterable to a list
+        onChanged: (value) {
+          //Sets the onChanged callback
+          if (value == null) {
+            return; //Returns if the value is null
+          }
+          setState(() {
+            _selectedCategory =
+                value; //Sets the selected category and updates the widget
+          });
+        });
+
+    var dateSelector = Expanded(
+      child: Row(
+        mainAxisAlignment: MainAxisAlignment.end,
+        crossAxisAlignment: CrossAxisAlignment.center,
+        children: [
+          Text(
+            _selectedDate == null //Displays the selected date or a placeholder
+                ? "No date selected"
+                : formatter.format(_selectedDate!),
+          ),
+          IconButton(
+            onPressed: _presentDatePicker,
+            icon: const Icon(Icons.calendar_month_outlined),
+          ),
+        ],
+      ),
+    );
+
+    var cancelButton = TextButton(
+      onPressed: () {
+        Navigator.pop(context); //Closes the overlay
+      },
+      child: const Text(
+        "Cancel",
+      ),
+    );
+
+    var confirmButton = ElevatedButton(
+        onPressed: _saveExpenseData,
+        child: const Text('Save')); //Creates the confirm button
+
+    final keyBoardSize =
+        MediaQuery.of(context).viewInsets.bottom; //Gets the keyboard size
+    return LayoutBuilder(builder: (ctx, constraints) {
+      //Creates a layout builder
+      final width = constraints.maxWidth; //Gets the max width
+
+      return SizedBox(
+        height:
+            double.infinity, //Makes the widget take up all the available space
+        child: SingleChildScrollView(
+          child: Padding(
+            padding: EdgeInsets.fromLTRB(16, 16, 16,
+                keyBoardSize + 16), //Adds padding and avoids the keyboard
+            child: Column(children: [
+              if (width >= 600) //Checks if the screen is wider than 600 pixels
```

Comment:
You could have used only one check for `width >= 600`. Is screen wide - describe entire layout for it. Not super wide - entire layout for smaller width.  
This would make the return statement more readable and less prone for logic errors (I mean forgetting to add a button in one of the views, etc).